fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -52,31 +63,43 @@ def fit(self, X, Y, **kwargs):
         if self._KW_NUMBER_LAGRANGE_MULTIPLIERS in kwargs:
             number_of_lagrange_multipliers = kwargs[self._KW_NUMBER_LAGRANGE_MULTIPLIERS]
 
+        # Extract the target attribute
+        A = self._make_vector(aux_data, "aux_data")
+
+        # Extract the Y values
+        Y_vector = self._make_vector(Y, "Y")
+
+        X_rows, _ = self._get_matrix_shape(X, "X")
+        if X_rows != Y_vector.shape[0]:
+            raise RuntimeError(self._MESSAGE_X_Y_ROWS)
+        if X_rows != A.shape[0]:
+            raise RuntimeError(self._MESSAGE_X_A_ROWS)
+
         # Prep the quality metric
-        self.quality_metric.set_data(X, Y, aux_data)
+        self.quality_metric.set_data(X, Y_vector, A)
 
         # We do not yet have disparity metrics fully implemented
         # For now, we assume that if we are passed a DemographicParity
         # object we have a binary classification problem whereas
         # BoundedGroupLoss indicates a regression
         if isinstance(self.disparity_metric, DemographicParity):
-            self._fit_classification(X, Y, aux_data,
+            self._fit_classification(X, Y_vector, A,
                                      lagrange_multipliers, number_of_lagrange_multipliers)
         elif isinstance(self.disparity_metric, BoundedGroupLoss):
-            self._fit_regression(X, Y, aux_data,
+            self._fit_regression(X, Y_vector, A,
                                  lagrange_multipliers, number_of_lagrange_multipliers)
         else:
             raise RuntimeError("Can't get here")
 
-    def _fit_classification(self, X, Y, protected_attribute,
+    def _fit_classification(self, X, Y, target_attribute,
```

Comment:
Mindful of 'protected attribute' having legal meaning, I've been changing to `target_attribute` ….thoughts? This is for more than just binning, so I think this is more descriptive